### PR TITLE
Store message drafts with secure storage

### LINKED ISSUE #482: Store Drafts
When user types a message and leaves a chat draft should be preserved. When user goes back to that chat what he wrote should still be in the message input box.

### PR DESCRIPTION
**Pull request by untreu2**
## Description
Added a service for managing draft messages with secure storage.

- Text is automatically saved as draft 500ms after user stops typing. 
- When user enters a chat, any existing draft is loaded into the input field. 
- Draft is automatically cleared when message is sent. 
- Drafts are not saved when editing existing messages.
- Immediately save draft when app is paused/minimized or becomes inactive

https://github.com/user-attachments/assets/83f1aceb-b38f-4165-bcae-34ec85e870a6

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore
- [ ] 🧪 Tests

## Checklist

- [x] Run `just precommit` to ensure that formatting and linting are correct
- [ ] Run `just check-flutter-coverage` to ensure that flutter coverage rules are passing
- [ ] Updated the `CHANGELOG.md` file with your changes (if they affect the user experience)

Fixes #482 

## Summary by CodeRabbit

* **New Features**
  * Per-chat draft auto-save (debounced while typing and on app pause), restore on return, persists across restarts; drafts cleared after sending or when emptied; global "clear all drafts" run during logout.

* **Tests**
  * Unit tests covering save/load/clear behaviors and edge cases.
  * In-memory secure-storage test helper and generated mocks for storage.

### CODE REVIEW

**Comment by Quwaysim** [comment]
I suppose this will be updated as per the lifecycle discussion? @untreu2

**Comment by untreu2** [comment]
Yes @Quwaysim

**Review by josefinalliende — COMMENTED** [review]
I tested it locally and it works great! 🚀 Really nice feature 💯 I just left two small comments.

One suggestion: it’d be awesome to add tests for `DraftMessageService`. I drafted a proposal with dependency injection in a branch called` test-store-drafts `feel free to cherry-pick that commit if you like it!

**Comment by josefinalliende** [comment]
> I tested it locally and it works great! 🚀 Really nice feature 💯 I just left two small comments.
> 
> One suggestion: it’d be awesome to add tests for `DraftMessageService`. I drafted a proposal with dependency injection in a branch called`test-store-drafts`feel free to cherry-pick that commit if you like it!

Opened a draft here so it's easier to take a look to the test  proposed: https://github.com/parres-hq/whitenoise_flutter/pull/579

File: lib/domain/services/draft_message_service.dart [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  static Future<void> clearDraft({required String chatId}) async {
+    try {
+      final key = '$_draftPrefix$chatId';
+      await _storage.delete(key: key);
+    } catch (e) {
+      return;
+    }
+  }
+
+  static Future<void> clearAllDrafts() async {
```
[the remark this answers is not in the document]
Comment (untreu2) [reply lib/domain/services/draft_message_service.dart:41]: Yes. It is what it is for.

**Review by josefinalliende — COMMENTED** [review]
I tested locally in ios emulator again after the changes — closed the app, reopened it, and drafts look good! ✅ Works perfectly for me 💯

My earlier comments still apply. TL;DR:
- The service has no logger — worth discussing if we want to use one here or not, just to stay consistent.
- In logout, maybe clearAllDrafts (currently unused) could be applied.
- I added a proposal for tests for the draft message service.

**Comment by Quwaysim** [comment]
> I tested locally in ios emulator again after the changes — closed the app, reopened it, and drafts look good! ✅ Works perfectly for me 💯

My earlier comments still apply. TL;DR:

The service has no logger — worth discussing if we want to use one here or not, just to stay consistent.
In logout, maybe clearAllDrafts (currently unused) could be applied.
I added a proposal for tests for the draft message service.

For the Logger @josefinalliende I think we don't need it here and we should try to minimise its usage so as not to clog the debug console.

**Review by Quwaysim — APPROVED** [review]
LGTM
